### PR TITLE
Use a different screenshot folder for Deadman mode

### PR DESCRIPTION
**Pull request by Nightfirecat**
Code based largely on #2098.

Closes runelite/runelite#2050

### CODE REVIEW

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 1]
```diff
@@ -401,9 +416,34 @@ public void hideWidgets(WidgetHiddenChanged event)
...
 
+	@Subscribe
+	public void onGameStatechange(GameStateChanged gameStateChanged)
+	{
+		if (worldResult == null)
+		{
+			return;
+		}
+
+		final World world = worldResult.findWorld(client.getWorld());
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: You dont need to do this, there is client.getWorldTypes

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: Good point, removed the dependency on `WorldResult` and switched to using `client.getWorldType` on game state change. :+1:

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 2]
```diff
@@ -401,9 +406,29 @@ public void hideWidgets(WidgetHiddenChanged event)
...
 
+	@Subscribe
+	public void onGameStatechange(GameStateChanged event)
+	{
+		if (event.getGameState() != GameState.LOGGED_IN)
+		{
+			return;
+		}
+
+		final EnumSet<WorldType> worldTypes = client.getWorldType();
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: You dont need to do this onGameStateChange now at all

File: runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java [hunk 3]
```diff
@@ -474,7 +481,17 @@ void takeScreenshot(String fileName)
 			File playerFolder;
 			if (client.getLocalPlayer() != null && client.getLocalPlayer().getName() != null)
 			{
-				playerFolder = new File(SCREENSHOT_DIR, client.getLocalPlayer().getName());
+				final EnumSet<WorldType> worldTypes = client.getWorldType();
+				final boolean dmm = worldTypes.contains(WorldType.DEADMAN);
+				final boolean sdmm = worldTypes.contains(WorldType.SEASONAL_DEADMAN);
+				final boolean IS_DMM_WORLD = dmm || sdmm;
```
Comment (deathbeam) [inline runelite-client/src/main/java/net/runelite/client/plugins/screenshot/ScreenshotPlugin.java]: rename the variable to not look like constant

**Comment by Adam-** [comment]
What is with the config reordering here?

**Comment by Nightfirecat** [comment]
I reordered the source to match the order of configs. (current code has `hotkey` and `screenshotKills` out of order compared to how they're displayed)

I can revert that and just set `screenshotKills` to position 8 and add `enableDateInFilename` at the end of the file with position 9 instead if you'd like.

**Comment by Adam-** [comment]
Fixed in 6c803dce035d3f18e179421bedc36e7df1ea7826

**Pull request closed without merging**